bin/runtime-common/src/messages_xcm_extension.rs

```diff
@@ -46,16 +53,25 @@ pub enum XcmBlobMessageDispatchResult {
 }
 
 /// [`XcmBlobMessageDispatch`] is responsible for dispatching received messages
-pub struct XcmBlobMessageDispatch<DispatchBlob, Weights> {
-	_marker: sp_std::marker::PhantomData<(DispatchBlob, Weights)>,
+///
+/// It needs to be used at the target bridge hub.
+pub struct XcmBlobMessageDispatch<DispatchBlob, Weights, Channel> {
+	_marker: sp_std::marker::PhantomData<(DispatchBlob, Weights, Channel)>,
 }
 
-impl<BlobDispatcher: DispatchBlob, Weights: MessagesPalletWeights> MessageDispatch
-	for XcmBlobMessageDispatch<BlobDispatcher, Weights>
+impl<
+		BlobDispatcher: DispatchBlob,
+		Weights: MessagesPalletWeights,
+		Channel: XcmChannelStatusProvider,
+	> MessageDispatch for XcmBlobMessageDispatch<BlobDispatcher, Weights, Channel>
 {
 	type DispatchPayload = XcmAsPlainPayload;
 	type DispatchLevelResult = XcmBlobMessageDispatchResult;
 
+	fn is_active() -> bool {
+		!Channel::is_congested()
```

Comment:
This code is executed at the target bridge hub. I suggest to use the following code for the `Channel` implementation:
```rust
pub struct LocalXcmChannelWithSiblingAssetHub;

impl bp_xcm_bridge_hub_router::XcmChannelStatusProvider for LocalXcmChannelWithSiblingAssetHub {
	fn is_congested() -> bool {
		// let's find the channel with the sibling AH
		let sibling_asset_hub_id: cumulus_primitives_core::ParaId = xcm_config::SiblingAssetHubParId::get().into();
		let outbound_channels = cumulus_pallet_xcmp_queue::OutboundXcmpStatus::<Runtime>::get();
		let channel_with_sibling_asset_hub = outbound_channels.iter()
			.filter(|c| c.recipient() == sibling_asset_hub_id)
			.next();

		// no channel => it is empty, so not congested
		let channel_with_sibling_asset_hub = match channel_with_sibling_asset_hub {
			Some(channel_with_sibling_asset_hub) => channel_with_sibling_asset_hub,
			None => return false,
		};

		// suspended channel => it is congested
		if channel_with_sibling_asset_hub.is_suspended() {
			return true;
		}

		// TODO: the following restriction is arguable, we may live without that, assuming that there
		// can't be more than some `N` messages queued at the bridge queue (at the source BH) AND before
		// accepting next (or next-after-next) delivery transaction, we'll receive the suspension signal
		// from the target AH and stop accepting delivery transactions

		// it takes some time for target AH to suspend inbound channel with the target BH and during that
		// we will keep accepting new message delivery transactions. Let's also reject new deliveries if
		// there are too many "pages" (concatenated XCM messages) in the target BH -> target AH queue.
		const MAX_QUEUED_PAGES_BEFORE_DEACTIVATION: u16 = 4;
		if channel_with_sibling_asset_hub.queued_pages() > MAX_QUEUED_PAGES_BEFORE_DEACTIVATION {
			return true;
		}

		true
	}
}
```